src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,12 +12,18 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = 
+            "Person address should consists of Block, Street, Unit, and Postal Code";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
     private boolean isPrivate;
+    private String value;
```

Comment:
there is no point maintaining 'value' anymore. It is duplicating data and could lead to data inconsistencies.